src/database/index.tsx

```diff
@@ -397,7 +396,17 @@ export const saveFacility = async (
     if (facility.id) {
       const payload = buildUpdatePayload(facility);
       facilityDoc = facilitiesCollection.doc(facility.id);
-      batch.update(facilityDoc, payload);
+      // Only update the facility if the incoming observedAt date is > than the current updatedAt date in the existing facility
+      const incomingObservedAt = facility.modelInputs.observedAt;
+      const currentFacilityData = await facilityDoc.get();
+      const currentFacility = buildFacility(scenarioId, currentFacilityData);
+      if (
+        incomingObservedAt &&
+        currentFacility.modelInputs.updatedAt &&
+        incomingObservedAt > currentFacility.modelInputs.updatedAt
```

Comment:
Case in point for that first issue, I think I just ran into this issue with my own facility model. 

So for example, I have a row that has a facility with `modelInputs.updatedAt` and `modelInputs.observedAt` dates on April 29th (around 4:30pm local time, to be precise, with `updatedAt` one second after `observedAt`).

If you open up the modal on a facility row, it pre-populates the observed at date with the current value.

So when I click on the number to open the modal it pre-populates the "Observed at" date picker with April 29th. If I update a value and click save, nothing happens.

This would _appear_ to be because of the logic I outlined in case 1, because that incoming `observedAt` date is probably _either_ 1) the beginning of the day (and thus hours earlier than the most recent `updatedAt` date) or 2) an unchanged value of the existing `modelInputs.observedAt` date (and thus still one second before the most recent `updatedAt`). Either way, I think users would find that very surprising. I would think they'd expect to be able to reenter observed data for a given day and see it updated.

It makes me wonder if the right solution here is to simply change from an > check to an >= check, and reduce the precision of the check to the day level?

Reply:
Fortunately, from my testing, it does appear in these cases that we do correctly persist an addition to `modelVersions` with the appropriate `observedAt` timestamp, so that's good!

Reply:
Ah I see. The problem is because the datepicker is only a day level precision. In that case, the datepicker gives a midnight timestamp for observedAt, while the current updatedAt is a precise ts.
I think changing this check to day precision level makes the most sense. Also, in that case it would be enough to compare the incoming observedAt to the current observedAt (instead of updatedAt).